### PR TITLE
Filter searches by region/country slugs

### PR DESCRIPTION
**Pull request by joel-wright**
This PR does the following:
- search by country/region slug
- expands region to constituent countries
- maps slug -> country code in search request

### CODE REVIEW

File: backend/app/api/api_v1/routers/search.py [hunk 1]
```diff
@@ -76,3 +86,37 @@ def search_documents(
...
+
+    for field, values in request_filters.items():
+        if field == FilterField.REGION:
+            field = FilterField.COUNTRY
+            filter_values = []
+            for geo_slug in values:
+                filter_values.extend(
+                    [g.value for g in get_countries_for_region(db, geo_slug)]
+                )
+            print(f"FILTER VALUES: {filter_values}")
```
Comment (diversemix) [inline backend/app/api/api_v1/routers/search.py:105]: rouge print?

File: backend/app/core/lookups.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+def get_countries_for_region(db: Session, region_slug: str) -> Sequence[Geography]:
+    geography = db.query(Geography).filter(Geography.slug == region_slug).first()
+    if geography is None:
+        return []
+
+    is_valid_region = geography.parent_id is None
+    if not is_valid_region:  # either unknown or not a region
+        return []
+
+    cast(Geography, geography)
```
Comment (diversemix) [inline backend/app/core/lookups.py:50]: Why the floating `cast` ?

**Review by diversemix — APPROVED** [review]
Looks good - one question,  one print statement

File: backend/app/api/api_v1/routers/search.py (hunk 1 above)
Comment (joel-wright) [reply backend/app/api/api_v1/routers/search.py:105]: AGH!

File: backend/app/core/lookups.py (hunk 2 above)
Comment (joel-wright) [reply backend/app/core/lookups.py:50]: type checker doesn't believe you otherwise

File: backend/app/api/api_v1/routers/search.py (hunk 1 above)
Comment (joel-wright) [reply backend/app/api/api_v1/routers/search.py]: sorted

File: backend/app/core/lookups.py (hunk 2 above)
Comment (diversemix) [reply backend/app/core/lookups.py:50]: Ah! ... understood, thanks :+1:

**Pull request merged**